### PR TITLE
[EasyCore] hotfix/open-api-normalizer

### PR DESCRIPTION
**Pull request by alexandrtspl**
* openApiNormalizer

| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes 
| BC breaks?    | no     
| Deprecations? | no 
| Tests pass?   | yes    
| Fixed tickets | #...

### CODE REVIEW

File: packages/EasyCore/src/Bridge/Symfony/ApiPlatform/OpenApi/Processor/DefaultDecorationProcessorInterface.php [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<?php
+
+declare(strict_types=1);
+
+namespace EonX\EasyCore\Bridge\Symfony\ApiPlatform\OpenApi\Processor;
+
+interface DefaultDecorationProcessorInterface
```
Comment (alexndlm) [inline packages/EasyCore/src/Bridge/Symfony/ApiPlatform/OpenApi/Processor/DefaultDecorationProcessorInterface.php:7]: I can't get why we need this interface?

Comment (alexndlm) [reply packages/EasyCore/src/Bridge/Symfony/ApiPlatform/OpenApi/Processor/DefaultDecorationProcessorInterface.php:7]: Got it

File: packages/EasyCore/src/Bridge/Symfony/ApiPlatform/OpenApi/Processor/UpdateDocsComponentsSchemasProcessor.php [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        unset($property);
+    }
+
+    /**
+     * @param mixed[] $properties
+     */
+    private function updateDocsPropertyLinks(array &$properties): void
+    {
+        $property = [];
+        $dateTimeExample = CarbonImmutable::now()->toIso8601String();
```
Comment (alexndlm) [inline packages/EasyCore/src/Bridge/Symfony/ApiPlatform/OpenApi/Processor/UpdateDocsComponentsSchemasProcessor.php:69]: Let's use here some exact date to make generated docs the same.

And let's add tests:
- two API resources, with all cases that we transform in our transformers.
- assert that generated JSON is the same.

Maybe we should think about creating the new package `easy-api-platform`, and move all related stuff there.

File: packages/EasyCore/src/Bridge/Symfony/ApiPlatform/OpenApi/Provider/OpenApiContextProvider.php [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+                            new Assert\NotBlank(),
+                        ],
+                        'schema' => [
+                            new Assert\Type('array'),
+                        ],
+                        'collection' => $itemConstraintsOptional,
+                        'item' => $itemConstraintsOptional,
+                    ],
+                    'allowExtraFields' => true,
+                    'allowMissingFields' => true,
```
Comment (alexndlm) [inline packages/EasyCore/src/Bridge/Symfony/ApiPlatform/OpenApi/Provider/OpenApiContextProvider.php:186]: Looks that this must be `false`?

**Review by alexndlm — CHANGES_REQUESTED** [review]

**Review by natepage — APPROVED** [review]

**Comment by natepage** [comment]
Add comment after approving review

**Review by natepage — APPROVED** [review]
PLACEHOLDER

**Review by natepage — APPROVED** [review]

**Review by natepage — APPROVED** [review]

**Review by natepage — APPROVED** [review]

**Review by natepage — APPROVED** [review]

**Review by natepage — APPROVED** [review]

**Review by natepage — APPROVED** [review]

**Review by natepage — APPROVED** [review]
Nice work! Code review complete.

- [X] Functionality and requirements
- [X] Code organization and structure
- [X] Code readability and maintainability
- [X] Error handling and exception management
- [X] Security considerations (POL-015)

**Review by natepage — APPROVED** [review]
Nice job! 😊 Code review passed.

- [X] Functionality and requirements
- [X] Code organization and structure
- [X] Code readability and maintainability
- [X] Error handling and exception management
- [X] Security considerations (POL-015)

**Review by natepage — APPROVED** [review]
Code review successful. Nice work!

- [X] Functionality and requirements
- [X] Code organization and structure
- [X] Code readability and maintainability
- [X] Error handling and exception management
- [X] Security considerations (POL-015)

**Pull request closed without merging**